backend/src/routes/content.rs

```diff
@@ -0,0 +1,413 @@
+                modified_time: Set(modified_time.with_timezone(&FixedOffset::east(8 * 3600))),
+                content: Set(Some(content.content.to_string()).to_owned()),
+                ..Default::default()
+            };
+            // insert the row in database
+            let res1 = content_reply
+                .insert(&pg_con)
+                .await
+                .expect("Cannot save content");
+            // println!("{:?}", res1.reply_id.unwrap());
+            reply_response.reply_id = subject_info.post_len;
+            // modify the time and the post_len in content_subject
+            let mut subject_content: pgdb::content_subject::ActiveModel =
+                subject_info.into_active_model();
+            subject_content.modified_time =
+                Set(modified_time.with_timezone(&FixedOffset::east(8 * 3600)));
+            subject_content.post_len = Set(reply_response.reply_id + 1);
+            // insert the row in database
+            let res2 = subject_content
+                .update(&pg_con)
+                .await
+                .expect("Cannot update content");
+            println!("{:?}", res2.post_id.unwrap());
+            reply_response.success = true;
+            // return the response
+            (Status::Ok, Json(reply_response))
+        }
+    }
+}
+
+#[delete("/<post_id>/delete")]
+pub async fn content_delete_post(
+    db: Connection<PgDb>,
+    post_id: i32,
+) -> (Status, Json<PostResponse>) {
+    let pg_con = db.into_inner();
+    // create a response struct
+    let mut post_response = PostResponse {
+        success: false,
+        error: Vec::new(),
+        post_id: 0i32,
+    };
+    // check if the post not exsits, add corresponding error if so
+    let subject_info = ContentSubject::find_by_id(post_id)
+        .one(&pg_con)
+        .await
+        .expect("cannot fetch content from pgdb");
+    if None == subject_info {
+        post_response.error.push("Post not exsits".to_string());
+        (Status::BadRequest, Json(post_response))
+    } else {
+        /* TODO */
+        // check if time is within limit, if so, allow user to delete
+        // delete data in content_subject
+        let subject_info: pgdb::content_subject::ActiveModel = subject_info.unwrap().into();
+        subject_info
+            .delete(&pg_con)
+            .await
+            .expect("cannot delete content from content_subject");
+        // delete data in content_reply
+        ContentReply::delete_many()
+            .filter(pgdb::content_reply::Column::PostId.eq(post_id))
+            .exec(&pg_con)
+            .await
+            .expect("cannot delete content from content_reply");
+        // return the response
+        post_response.success = true;
+        post_response.post_id = post_id;
+        (Status::Ok, Json(post_response))
+    }
+}
+
+#[delete("/<post_id>/<reply_id>/delete")]
+pub async fn content_delete_reply(
```

Comment:
change uri to "/post/<post_id>/reply/<reply_id>", information can be carried in json data body